README.md

```diff
@@ -0,0 +1,269 @@
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+   Technologies and Tools
+   Key Technologies
+   Java Development Kit (JDK) 17
+   Maven
+   MySQL
+   Spring Boot
+   H2 Database (for testing)
+   Lombok
+   MapStruct
+   Liquibase
+   SpringDoc OpenAPI
+   JWT (JSON Web Token)
+   Docker Compose
+   
+
+
+4. **Access the application**:
+    - The application runs on `http://localhost:8080`.
+    - API documentation is available at `http://localhost:8080/swagger-ui.html`.
+
+
+
+### Docker Setup
+
+**Swagger**:
+   - You can use Swagger for manipulating API endpoints.
+
+**Docker**:
+   - You can download the container from Docker and start the API.
+
+*** docker-compose
+
+services:
+mysqldb:
+restart: always
+env_file: ./.env
+image: mysql:8.0
+environment:
+MYSQL_ROOT_PASSWORD: ${DB_PASSWORD}
+MYSQL_DATABASE: ${MYSQL_DATABASE}
+MYSQL_USER: ${MYSQL_USER}
+MYSQL_PASSWORD: ${MYSQL_PASSWORD}
+ports:
+- "3309:3306"
+volumes:
+- db_data:/var/lib/mysql
+healthcheck:
+test: [ "CMD-SHELL", "mysqladmin ping -h localhost -P ${DB_DOCKER_PORT} -u ${DB_USER} -p${DB_PASSWORD}" ]
+interval: 30s
+timeout: 10s
+retries: 3
+start_period: 60s
+
+app:
+depends_on:
+mysqldb:
+condition: service_healthy
+image: task-manager-api
+build: .
+env_file: ./.env
+ports:
+- "${SPRING_LOCAL_PORT}:${SPRING_DOCKER_PORT}"
+- "${DEBUG_PORT}:${DEBUG_PORT}"
+environment:
+SPRING_APPLICATION_JSON: '{
+"spring.datasource.url": "jdbc:mysql://mysqldb:${DB_DOCKER_PORT}/${DB_DATABASE}?serverTimezone=UTC",
+"spring.datasource.username": "${DB_USER}",
+"spring.datasource.password": "${DB_PASSWORD}",
+"jwt.secret": "${JWT_SECRET}"
+}'
+JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5000"
+
+volumes:
+db_data:
+
+*** Dockerfile
+
+# Builder stage
+FROM openjdk:21-jdk-slim as builder
+WORKDIR application
+ARG JAR_FILE=target/*.jar
+COPY ${JAR_FILE} application.jar
+RUN java -Djarmode=layertools -jar application.jar extract
+
+# Final stage
+FROM openjdk:21-jdk-slim
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.launch.JarLauncher"]
+EXPOSE 8080
+
+*** env file
+
+# MySQL environment variables
+DB_PASSWORD=123sqlbod321
+DB_DATABASE=book_store_db
+DB_USER=bodya
+DB_LOCAL_PORT=3309
+DB_DOCKER_PORT=3306
+
+# Spring Boot application environment variables
+SPRING_LOCAL_PORT=8080
+SPRING_DOCKER_PORT=8080
+
+# Debug port
+DEBUG_PORT=5005
+
+# JWT secret key
+JWT_SECRET=wefwqfdfwqfq4f3fq3af43453452ef34fewfewfe
+
+MYSQL_DATABASE=book_store_db
+MYSQL_USER=bodya
+MYSQL_PASSWORD=123sqlbod321
```

Comment:
this is useful but it should be formatted properly, as a block of code and a single file
https://docs.github.com/ru/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks